tests/core/pyspec/eth2spec/test/gloas/block_processing/test_process_attestation.py

```diff
@@ -0,0 +1,458 @@
+    source_flag = spec.has_flag(final_participation, spec.TIMELY_SOURCE_FLAG_INDEX)
+    target_flag = spec.has_flag(final_participation, spec.TIMELY_TARGET_FLAG_INDEX)
+
+    # Verify the core functionality: attestation processes and gets some participation flags
+    assert source_flag or target_flag, (
+        "Should have participation flags when attestation processes successfully"
+    )
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_matching_payload_false_historical_slot(spec, state):
+    """
+    Test is_matching_payload = False path for historical slots
+    (when data.index does NOT match the payload availability bit)
+    """
+    apply_empty_block(spec, state, 1)
+    apply_empty_block(spec, state, 2)
+    apply_empty_block(spec, state, 3)
+    next_slots(spec, state, spec.MIN_ATTESTATION_INCLUSION_DELAY)
+
+    # Choose slot 2 which now has a real block root different from genesis
+    historical_slot = 2
+    availability_bit_index = historical_slot % spec.SLOTS_PER_HISTORICAL_ROOT
+
+    # Set payload availability bit to 0 for slot 2
+    state.execution_payload_availability[availability_bit_index] = 0
+
+    # Create attestation for slot 2 but with slot 1's block root to make it historical (not same-slot)
+    slot_1_block_root = spec.get_block_root_at_slot(state, 1)
+    attestation = get_valid_attestation(
+        spec, state, slot=historical_slot, beacon_block_root=slot_1_block_root
+    )
+    attestation.data.index = 1  # Does not match the availability bit (0)
+    sign_attestation(spec, state, attestation)
+
+    # Get the attesting validator
+    attesting_indices = spec.get_attesting_indices(state, attestation)
+    validator_index = list(attesting_indices)[0]
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is False
+
+    # This should still pass (the attestation is valid)
+    # but is_matching_payload will be False, affecting participation flags
+    yield from run_attestation_processing(spec, state, attestation, valid=True)
+
+    # Verify that head flag was NOT set due to mismatched payload
+    final_participation = state.current_epoch_participation[validator_index]
+    final_head_flag = spec.has_flag(final_participation, spec.TIMELY_HEAD_FLAG_INDEX)
+    assert not final_head_flag, "Should not get head flag when payload doesn't match"
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_matching_payload_gets_head_flag(spec, state):
+    """
+    Test processing attestation with matching payload for historical slots
+    """
+    # Advance to slot 3 (only creates one block at slot 3)
+    transition_to_slot_via_block(spec, state, 3)
+
+    # Move forward exactly 1 slot to get inclusion_delay == MIN_ATTESTATION_INCLUSION_DELAY (which is 1)
+    next_slots(spec, state, 1)  # Now at slot 4
+
+    # Choose slot 3 for the attestation so inclusion_delay = 4 - 3 = 1
+    # This should NOT be same-slot because we're processing at a different slot
```

Comment:
The attestation slot is still 3 so this *is* a same slot attestation, it's voting for the current slot at it's time which is slot 3 and the blockroot at that slot is different than at slot 2, so  it doesn't matter when it's processed.